### PR TITLE
add browse method

### LINKED ISSUE #2: Add browse method
Add browse method to plugin:

- be able to browse all insights (active/inactive for admin, active only for primary)
- be able to filter by a tag.

### PR DESCRIPTION
**Pull request by mk4111**
Closes #2 

Adds browse method with filter-by-tag option too.

Build fails because this PR relies on [insights to be added to mockdata](https://github.com/emfoundation/mock-data/pull/2)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,48 @@
...
+    title: 'Insight Number 1',
+    url: 'https://....',
+    tags: [
+    {
+      tag_id: 8,
+      tag_name: 'Agriculture'
+    }, {
+      tag_id: 22,
+      tag_name: 'Telecommunications'
+    }, {
```
Comment (jay-meister) [inline README.md:40]: I would imagine we want to return the activity status of the insight which will be necessary in the case when activeOnly is false

File: lib/formatters/browse.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+// helper function that takes a row and returns a insight object
+function insightFromRow (row) {
+  return {
+    id: row.iid,
+    title: row.title,
+    url: row.url,
+    author: row.author,
+    creator_id: row.creator_id,
+    date: row.date,
+    active: row.active,
```
Comment (jay-meister) [inline lib/formatters/browse.js:78]: active is indeed here, readme can be updated

File: lib/formatters/browse.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+   active: true,
+   resource: true },
+  anonymous {... ]
+
+WE WANT TO CONVERT THIS INTO:
+  [{
+  id: 1,
+  date: 2016-12-23T11:27:41.760Z,
+  title: 'Insight Number 1',
+  url: 'https://....',
```
Comment (jay-meister) [inline lib/formatters/browse.js:47]: this seems out of date, could possibly remove from this file if it is in the README

**Comment by jay-meister** [comment]
looks good!
screen shot 2017-01-05 at 11 49 13

Wireframes suggest we need to store the type of data being stored, eg report/article.  I don't think that is being stored in this branch? @mk4111 any thoughts?

Also the size of the piece? I don't know how we can get this unless the admin manually adds it to when adding the insight to the app.

**Comment by jay-meister** [comment]
Also I don't think we need to show the tags that are associated with each insight according to the above wireframe, but since it is already done, we can probably leave it in there.

File: lib/formatters/browse.js (hunk 3 above)
Comment (mk4111) [reply lib/formatters/browse.js:47]: Ok!

**Comment by mk4111** [comment]
@JMurphyWeb - I was basing the things that I thought were required on [the main description of this issue](https://github.com/emfoundation/ce100-app/issues/324), which suggests 'Type' --> From our meeting with them I totally thought this meant document type, so '.pdf', etc. but I might've been wrong - Alex may have meant 'Report', 'Article' here... Or maybe they are meant to be two separate fields...? :confused: This should be something we ask!

With the size of the document, I think the metadata from our request to AWS can tell us this?

We can remove tags from the options being removed if that's better! I'm happy with either :)

**Comment by jay-meister** [comment]
@mk4111 where do you anticipate needing to use the doctype piece of data? I am struggling to think of a use case. I think the type dropdown is linked to the type of insight that is being added (report/article/image...)

**Comment by jay-meister** [comment]
btw i am still unsure how we can get the size of the insight without making a load of requests or asking the admin to add it in.

**Comment by mk4111** [comment]
@JMurphyWeb - I wasn't sure but I remembered it was mentioned in the meeting me had, but they very well may have just meant that they would want to be able to host xyz file types. From the main description it wasn't overly clear but I assumed that that was what they had meant - I may be wrong though! We can definitely sub `doctype` for `type` instead.

Yeah, I think you are right in that we would need to make a bunch of requests to get the filesize. It wasn't something that Alex had mentioned in the main description though, and I'm wondering if maybe this was added by Fanny on the assumption that users would be downloading something onto their local drive, when I think they will just be accessing it remotely via url - in which case do we need filesize? We should ask Alex about this too methinks!

**Comment by jay-meister** [comment]
yeah ok I will swap out the doctype for type.

Ok good point, i have asked alex regarding the file size so lets leave it with him.

**Comment by nelsonic** [comment]
Looking good so far. 👍

File: lib/queries/getByTag.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+/*
+* We pass getByTag a table that we want get ('challenges') and a tag id
+* that we want to filter by. This function should return all of the challenges
+* that meet that contain that tag.
+*/
+
+module.exports = function (tagId, activeOnly) {
+  /* FIND ALL CHALLANGES BY TAG ID THEN GET ALL TAGS FOR THOSE CHALLANGES */
```
Comment (SimonLab) [inline lib/queries/getByTag.js:10]: I think this function get the insights by tagId not the challenges

**Review by SimonLab — APPROVED** [review]
Just a few small changes that we can update on the next PR (update Readme and comments in the code)

Thanks @mk4111 @JMurphyWeb

**Pull request merged**